### DISCUSSION TITLE
InvalidStoredEvent When Retrieving Aggregate if Model Deleted...

### DISCUSSION THREAD

**Discussion by garethdaine** [spatie/laravel-event-sourcing#311] [Q&A] [answered]
Hey Folks,

Not entirely sure whether this is an actual bug or not, but I have an issue retrieving an aggregate when a model that was stored with an event on the aggregate has been deleted and no longer exists.

For example, we have a LeagueAggregate and users of the system can join leagues which are linked via a pivot table. When a LeagueJoined event is fired and stored, the user and league models (IDs) are stored in the event properties.

If a user is later deleted from the system, trying to retrieve that aggregate will now always return an **InvalidStoredEvent** exception.

So, if future users try to join that league, they'll receive that exception and the system will break.

Any ideas on the best way to handle this? Thanks.

**Comment by etahamer** [comment, marked as answer]
Hey @garethdaine,

So your event constructor looks something similar?

![image](https://user-images.githubusercontent.com/8310837/150123515-228e4ed5-10ab-40c0-aa25-299cef31e3e4.png)

If yes, then I suggest not storing model as a constructor argument and rather store user_uuid property instead. This way you could always retrieve aggregate and also replay events without depending on database.

**Reply by brendt** [reply]
This is the right answer: don't store a model on the event.

**Reply by garethdaine** [reply]
Thanks @brendt and etahamer. Very much appreciated.

I hadn't realised you shouldn't store models on the event. Is this mentioned in the docs anywhere at all? Might also be good for a future update/version to prevent this.

Quick question, folks, how would you approach fixing already stored events?

I'm thinking of creating a console command that updates all stored events that previously had models stored, and replacing with the UUID. Once the aggregate/event code has been updated, of course.

Or, is there a better solution?

**Reply by brendt** [reply]
In this case I'd write a migration that actually changes the JSON payload in the stored events, since it's an issue with a rather technical nature.
